shopfloor_mobile_base/static/wms/src/main.js

```diff
@@ -136,26 +135,25 @@ new Vue({
             let params = _.defaults({}, odoo_params, {
                 debug: this.demo_mode,
                 base_url: this.app_info.base_url,
-                // TODO: move out to its own handler
-                // when full aut decoupling happens
-                headers: {
-                    "API-KEY": this.apikey,
-                },
             });
             let OdooClass = null;
             if (this.demo_mode) {
                 OdooClass = OdooMocked;
             } else {
                 OdooClass = Odoo;
             }
+            const auth_handler = this._get_auth_handler();
+            params = _.merge({}, params, auth_handler.get_params(this));
+            // TODO: allow auth_handler to return OdooClass?
+            return new OdooClass(params);
+        },
+        _get_auth_handler: function (force) {
             const auth_type = this.app_info.auth_type;
             const auth_handler = auth_handler_registry.get(auth_type);
             if (_.isUndefined(auth_handler)) {
                 throw "Auth type '" + auth_type + " not supported";
```

Comment:
I know this is not part to this current work but I suppose we could fix that in the mean time? 
```suggestion
                throw "Auth type '" + auth_type + "' not supported";
```

Reply:
For record I were getting this error before installing  shopfloor_mobile_base_auth_api_key, to avoid loosing time we could suggested developer to ensure one of shopfloor mobile auth module is properly installed !

```suggestion
                throw "Auth type '" + auth_type + "' not supported, did you install the proper shopfloor auth module ?";
```